front/app/components/UI/ProjectFilter/index.tsx

```diff
@@ -103,25 +109,72 @@ const ProjectFilter = ({
     localize,
   ]);
 
+  // Filter options based on search value
+  const filteredOptions = useMemo(() => {
+    if (!allProjectOptions) return [];
+    if (!searchValue) return allProjectOptions;
+
+    return allProjectOptions.filter((option) =>
+      option.label.toLowerCase().includes(searchValue.toLowerCase())
+    );
+  }, [allProjectOptions, searchValue]);
+
+  const handleInputChange = useMemo(() => {
+    return debounce((searchTerm: string) => {
```

Comment:
Instead of using the `debounce` method, you can also try the `useDebouncedValue` hook that was introduced a while back. I think you could just do something like

```
const [searchValue, setSearchValue] = useDebouncedValue('', 300);
...
onInputChange={setSearchValue}